install.sh

```diff
@@ -0,0 +1,330 @@
+	apachefile=/etc/apache2/sites-available/$fqdn.conf
+	sudo apt-get update ; sudo apt-get -y upgrade ;	sudo apt-get install -y git unzip
+
+	if [[ $ver == "1" ]]; then
+		sudo git clone https://github.com/snipe/snipe-it.git $dir
+	else
+		wget https://github.com/snipe/snipe-it/archive/2.0.zip
+		sudo unzip 2.0.zip -d /var/www/
+	fi
+
+	#We already established MySQL root & user PWs, so we dont need to be prompted. Let's go ahead and install Apache, PHP and MySQL.
+	sudo DEBIAN_FRONTEND=noninteractive apt-get install -y lamp-server^
+	sudo apt-get install -y php5 php5-mcrypt php5-curl php5-mysql
+
+	#Create MySQL accounts
+	echo "Create MySQL accounts"
+	sudo mysqladmin -u root password $mysqlrootpw
+	sudo mysql -u root -p$mysqlrootpw < /root/createstuff.sql
+
+	#Enable mcrypt and rewrite
+	sudo php5enmod mcrypt
+	sudo a2enmod rewrite
+	sudo ls -al /etc/apache2/mods-enabled/rewrite.load
+
+	#Create a new virtual host for Apache.
+	echo >> $apachefile ""
+	echo >> $apachefile ""
+	echo >> $apachefile "<VirtualHost *:80>"
+	echo >> $apachefile "ServerAdmin webmaster@localhost"
+	echo >> $apachefile "    <Directory $dir/public>"
+	echo >> $apachefile "        Require all granted"
+	echo >> $apachefile "        AllowOverride All"
+	echo >> $apachefile "   </Directory>"
+	echo >> $apachefile "    DocumentRoot $dir/public"
+	echo >> $apachefile "    ServerName $fqdn"
+	echo >> $apachefile "        ErrorLog "\${APACHE_LOG_DIR}"/error.log"
+	echo >> $apachefile "        CustomLog "\${APACHE_LOG_DIR}"/access.log combined"
+	echo >> $apachefile "</VirtualHost>"
+	echo >> $hosts "127.0.0.1 $hostname $fqdn"
+	a2ensite $fqdn.conf
+
+	#Change permissions on directories
+	sudo chmod -R 755 $dir/app/storage
+	sudo chmod -R 755 $dir/app/private_uploads
+	sudo chmod -R 755 $dir/public/uploads
+	sudo chown -R www-data:www-data /var/www/
+	echo "Finished permission changes."
+
+	#Modify the Snipe-It files necessary for a production environment.
+	replace "'www.yourserver.com'" "'$hostname'" -- $dir/bootstrap/start.php
+	cp $dir/app/config/production/database.example.php $dir/app/config/production/database.php
+	replace "'snipeit_laravel'," "'snipeit'," -- $dir/app/config/production/database.php
+	replace "'travis'," "'snipeit'," -- $dir/app/config/production/database.php
+	replace "            'password'  => ''," "            'password'  => '$mysqluserpw'," -- $dir/app/config/production/database.php
+	replace "'http://production.yourserver.com'," "'http://$fqdn'," -- $dir/app/config/production/database.php
+	cp $dir/app/config/production/app.example.php $dir/app/config/production/app.php
+	replace "'http://production.yourserver.com'," "'http://$fqdn'," -- $dir/app/config/production/app.php
+	replace "'Change_this_key_or_snipe_will_get_ya'," "'$random32'," -- $dir/app/config/production/app.php
+	cp $dir/app/config/production/mail.example.php $dir/app/config/production/mail.php
+
+	#Install / configure composer
+	sudo curl -sS https://getcomposer.org/installer | php
```

Comment:
No, some lines may be artifacts of when I was testing Relentlessly against
2 virtual boxes, unfortunately I may have missed a few spots of the script
to clean up :/

On Mon, Aug 3, 2015 at 12:27 PM, snipe <notifications@github.com> wrote:

> In install.sh
> <https://github.com/snipe/snipe-it/pull/1006#discussion_r36121290>:
>
> > +	echo "Finished permission changes."
> > +
> > +	#Modify the Snipe-It files necessary for a production environment.
> > +	replace "'www.yourserver.com'" "'$hostname'" -- $dir/bootstrap/start.php
> > +	cp $dir/app/config/production/database.example.php $dir/app/config/production/database.php
> > +	replace "'snipeit_laravel'," "'snipeit'," -- $dir/app/config/production/database.php
> > +	replace "'travis'," "'snipeit'," -- $dir/app/config/production/database.php
> > +	replace "            'password'  => ''," "            'password'  => '$mysqluserpw'," -- $dir/app/config/production/database.php
> > +	replace "'http://production.yourserver.com'," "'http://$fqdn'," -- $dir/app/config/production/database.php
> > +	cp $dir/app/config/production/app.example.php $dir/app/config/production/app.php
> > +	replace "'http://production.yourserver.com'," "'http://$fqdn'," -- $dir/app/config/production/app.php
> > +	replace "'Change_this_key_or_snipe_will_get_ya'," "'$random32'," -- $dir/app/config/production/app.php
> > +	cp $dir/app/config/production/mail.example.php $dir/app/config/production/mail.php
> > +
> > +	#Install / configure composer
> > +	sudo curl -sS https://getcomposer.org/installer | php
>
> Sudo shouldn't be needed to download composer, should it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/snipe/snipe-it/pull/1006/files#r36121290>.
>